### PR TITLE
STR-534: Remove the usage of reth fork in strata

### PR DESCRIPTION
**Pull request by evgenyzdanovich**
## Description

Remove the usage of reth fork in strata, as it's not used anymore by the prover.

### Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [x] Refactor
-   [ ] New or updated tests

## Checklist

-   [x] I have performed a self-review of my code.
-   [ ] I have commented my code where necessary.
-   [ ] I have updated the documentation if needed.
-   [ ] My changes do not introduce new warnings.
-   [ ] I have added tests that prove my changes are effective or that my feature works.
-   [ ] New and existing tests pass with my changes.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -158,34 +158,34 @@ reqwest = { version = "0.12.7", default-features = false, features = [
...
+reth-errors = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-ethereum-payload-builder = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-evm = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-evm-ethereum = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-exex = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-ipc = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-node-api = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-node-ethereum = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-payload-builder = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3" }
+reth-primitives = { git = "https://github.com/paradigmxyz/reth.git", rev = "v1.0.3", default-features = false, features = [
```
Comment (storopoli) [inline Cargo.toml:176]: These don't need to be a `git` variant.
We can use crates.io and just put the version.
Example:

```toml
reth = 1.0.3
```

**Review by storopoli — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (evgenyzdanovich) [reply Cargo.toml:176]: TBH this was my initial intent, but strangely enough, it only has one release with 1.0.0 on crates.io: 
[reth crate](https://crates.io/crates/reth)

Or am I missing something?

**Review by storopoli — APPROVED** [review]
ACK e6a7477709baf62ee8f03fa1051111db9b777f71

This does not solve STR-534 but removes the fork so that everyone can have a happy weekend.

We'll work on bumping `reth` to latest along with `alloy` stuff

**Review by prajwolrg — APPROVED** [review]

**Comment by storopoli** [comment]
`alpenlabs/reth` was archived as well. Nice works guys!

**Pull request merged**